cmake/modules/SwiftUtils.cmake

```diff
@@ -123,3 +123,23 @@ function(set_with_default variable value)
     set(${variable} ${SWD_DEFAULT} PARENT_SCOPE)
   endif()
 endfunction()
+
+function(swift_create_post_build_symlink target is_directory path_to_file path_to_symlink working_directory comment)
```

Comment:
Rather than lots of named parameters, why not use `cmake_parse_arguments` with named parameters? That is much easier to read.

I'm fine with the source, dest, and target being named parameters, but `is_directory` should be a flag, and `working_directory` and `comment` should be single option parameters (potentially even optional).